src/Microsoft.OData.Core/ODataMessageReaderSettings.cs

```diff
@@ -235,6 +237,21 @@ public ODataMessageQuotas MessageQuotas
         /// </summary>
         internal bool ThrowOnUndeclaredPropertyForNonOpenType { get; private set; }
 
+        /// <summary>
+        /// Gets or sets a value that indicates whether the reader should put key values in their own URI segment when automatically building URIs.
+        /// If this value is false, automatically-generated URLs will take the form "../EntitySet('KeyValue')/..".
+        /// If this value is true, automatically-generated URLs will take the form "../EntitySet/KeyValue/..".
+        /// This setting only applies to URLs that are automatically generated by the <see cref="ODataMessageReader" /> and the URLs explicitly provided by the server won't be modified.
+        /// </summary>
+        public bool EnableReadingKeyAsSegment { get; set; }
+
+        /// <summary>
+        /// True if can read reserved annotation name without prefix 'odata.', otherwise false.

```

Comment:
```suggestion
        /// true to support reading reserved annotation name without prefix 'odata.', otherwise false.
```
nit: I think the practice is to use "true" instead of "True" in places where a boolean value is expected.